clients/include/testing_gemv_batched.hpp

```diff
@@ -0,0 +1,307 @@
+    rocblas_operation transA      = char2rocblas_operation(arg.transA);
+    rocblas_int       batch_count = arg.batch_count;
+
+    rocblas_local_handle handle;
+
+    //Device-arrays of pointers to device memory
+    T** dAA;
+    T** dxA;
+    T** dy_1A;
+    T** dy_2A;
+    hipMalloc(&dAA, batch_count * sizeof(T*));
+    hipMalloc(&dxA, batch_count * sizeof(T*));
+    hipMalloc(&dy_1A, batch_count * sizeof(T*));
+    hipMalloc(&dy_2A, batch_count * sizeof(T*));
+
+    // argument sanity check before allocating invalid memory
+    if(!dAA || !dxA || !dy_1A || !dy_2A)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    if(M < 0 || N < 0 || lda < M || lda < 1 || !incx || !incy)
+    {
+        EXPECT_ROCBLAS_STATUS(rocblas_gemv_batched<T>(handle,
+                                                      transA,
+                                                      M,
+                                                      N,
+                                                      &h_alpha,
+                                                      dAA,
+                                                      lda,
+                                                      dxA,
+                                                      incx,
+                                                      &h_beta,
+                                                      dy_1A,
+                                                      incy,
+                                                      batch_count),
+                              rocblas_status_invalid_size);
+
+        return;
+    }
+
+    size_t size_A = lda * static_cast<size_t>(N);
+    size_t size_x, dim_x, abs_incx;
+    size_t size_y, dim_y, abs_incy;
+
+    if(transA == rocblas_operation_none)
+    {
+        dim_x = N;
+        dim_y = M;
+    }
+    else
+    {
+        dim_x = M;
+        dim_y = N;
+    }
+
+    abs_incx = incx >= 0 ? incx : -incx;
+    abs_incy = incy >= 0 ? incy : -incy;
+
+    size_x = dim_x * abs_incx;
+    size_y = dim_y * abs_incy;
+
+    // Host-arrays of pointers to host memory
+    host_vector<T> hAA[batch_count];
+    host_vector<T> hxA[batch_count];
+    host_vector<T> hy_1A[batch_count];
+    host_vector<T> hy_2A[batch_count];
+    host_vector<T> hy_goldA[batch_count];
+
+    // Host-arrays of pointers to device memory
+    // (intermediate arrays used for the transfers)
+    T* AA[batch_count];
+    T* xA[batch_count];
+    T* y_1A[batch_count];
+    T* y_2A[batch_count];
+
+    for(int b = 0; b < batch_count; ++b)
+    {
+        hAA[b]      = host_vector<T>(size_A);
+        hxA[b]      = host_vector<T>(size_x);
+        hy_1A[b]    = host_vector<T>(size_y);
+        hy_2A[b]    = host_vector<T>(size_y);
+        hy_goldA[b] = host_vector<T>(size_y);
+        hipMalloc(&AA[b], size_A * sizeof(T));
+        hipMalloc(&xA[b], size_x * sizeof(T));
+        hipMalloc(&y_1A[b], size_y * sizeof(T));
+        hipMalloc(&y_2A[b], size_y * sizeof(T));
```

Comment:
`device_vector<T>` should be used instead of `hipMalloc`. It automatically frees resources. Please look at other test files for examples.

Reply:
Yes, I wanted to use device_vector from the beginning but could not get it to work. It compiles, but then seg-faults at run time...

Reply:
Under refactoring as per discussion

Reply:
New device_batch_vector class added to rocblas_vector.hpp